### PR TITLE
Upgrade dependencies and add spring-boot3

### PR DESCRIPTION
**Pull request by jrhee17**

### CODE REVIEW

File: gradle/scripts/build-flags.gradle [hunk 1]
```diff
@@ -35,7 +35,7 @@ if (!projectsWithFlags('java').isEmpty()) {
     }
 
     if (!projectsWithFlags('java', 'relocate').isEmpty()) {
-        ['com.github.jengelman.gradle.plugins:shadow', 'com.guardsquare:proguard-gradle'].each {
+        ['gradle.plugin.com.github.johnrengelman:shadow', 'com.guardsquare:proguard-gradle'].each {
```
Comment (jrhee17) [inline gradle/scripts/build-flags.gradle:38]: These changes have been introduced in armeria but never updated back upstream 😅 

https://github.com/line/gradle-scripts/pull/147

Comment (trustin) [reply gradle/scripts/build-flags.gradle:38]: Oopsie! Thanks, @jrhee17

**Review by trustin — APPROVED** [review]
:+1:

File: settings.gradle [hunk 2]
```diff
@@ -11,8 +11,10 @@ includeWithFlags ':client:java-armeria',                    'java', 'publish', '
 includeWithFlags ':client:java-armeria-legacy',             'java', 'publish', 'relocate'
 includeWithFlags ':client:java-spring-boot-autoconfigure',  'java', 'publish', 'relocate'
 includeWithFlags ':client:java-spring-boot-starter',        'java', 'publish', 'relocate'
-includeWithFlags ':client:java-spring-boot1-autoconfigure', 'java', 'publish', 'relocate'
-includeWithFlags ':client:java-spring-boot1-starter',       'java', 'publish', 'relocate'
+includeWithFlags ':client:java-spring-boot1-autoconfigure', 'java', 'publish', 'relocate', 'no_aggregation'
+includeWithFlags ':client:java-spring-boot1-starter',       'java', 'publish', 'relocate', 'no_aggregation'
+includeWithFlags ':client:java-spring-boot3-autoconfigure', 'java17', 'publish', 'relocate', 'no_aggregation'
+includeWithFlags ':client:java-spring-boot3-starter',       'java17', 'publish', 'relocate', 'no_aggregation'
```
Comment (ikhoon) [inline settings.gradle:17]: It is difficult to infer if `java-spring-boot-autoconfigure` means Spring Boot 2 or 3 from the name. What do you think of renaming `java-spring-boot-{autoconfigure,starter}` to `java-spring-boot2-{autoconfigure,starter}` although they are breaking changes?

For consistency, `':testing:junit'` needs to be `':testing:junit5'.

Comment (jrhee17) [reply settings.gradle]: Done 😄

File: build.gradle [hunk 3]
```diff
@@ -90,10 +90,12 @@ configure(projectsWithFlags('java')) {
...
-    tasks.withType(JavaCompile) {
-        if (JavaVersion.current() >= JavaVersion.VERSION_1_9) {
-            options.release = 8
+    // Target Java 8 except for spring-boot3
+    if (!project.name.startsWith("java-spring-boot3")) {
+        tasks.withType(JavaCompile) {
+            if (JavaVersion.current() >= JavaVersion.VERSION_1_9) {
+                options.release = 8
+            }
         }
```
Comment (ikhoon) [inline build.gradle:99]: Could we remove this block? According to what I understand, `--release` target is automatically set by `java.gradle`.

File: dependencies.toml [hunk 4]
```diff
@@ -14,6 +14,7 @@ mockito = "4.11.0"
 slf4j = "1.7.36"
 spring-boot1 = "1.5.22.RELEASE"
```
Comment (ikhoon) [inline dependencies.toml:15]: Should we remove this and its usage?
```suggestion
```

File: settings.gradle (hunk 2 above)
Comment (ikhoon) [reply settings.gradle]: - https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.7-Release-Notes#new-autoconfiguration-annotation
`META-INF/spring.factories` should be migrated into `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`
  It is weird our tests weren't broken without the migration 🤔 
- What do you think of making `java-spring-boot3*` the main source set and referring them to `java-spring-boot2*`.

File: build.gradle (hunk 3 above)
Comment (jrhee17) [reply build.gradle:99]: I ran into issues with jgit6 when I removed this previously 😅   Let me try again

Comment (jrhee17) [reply build.gradle:99]: I don't think this is a trivial issue 😅    Let me handle this in a separate issue and proceed with the release for now.

Comment (ikhoon) [reply build.gradle:99]: Sounds good.

File: settings.gradle (hunk 2 above)
Comment (jrhee17) [reply settings.gradle]: > It is weird our tests weren't broken without the migration 🤔

`@SpringBootApplication` enables `@ComponentScan` on the same package by default.
I've removed this configuration and confirmed that tests fail without the `spring.factories` metadata.

> What do you think of making java-spring-boot3* the main source set and referring them in java-spring-boot2*.

I have mixed feelings about this. Although I think it can be clear which module has the source set, I also think this is more work for little benefit (both for the reviewer and code submitter)

Having said this, I guess major version-ups don't occur often so I've updated for this iteration.

File: dependencies.toml (hunk 4 above)
Comment (jrhee17) [reply dependencies.toml]: 🙀  Done!

**Review by ikhoon — APPROVED** [review]
Thanks, @jrhee17! 🙇‍♂️
